app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -78,8 +89,39 @@ def member_object_ids
       ActiveFedora::Base.search_with_conditions("member_of_collection_ids_ssim:#{id}").map(&:id)
     end
 
+    protected
+
+      def collection_type=(collection_type)
+        @collection_type = collection_type
+      end
+
     private
 
+      # Load the collection_type attribute with an instance of the collection type of this collection based on the gid stored in the
+      # collection object model.  Defaults to the default collection type (i.e. user_collection) so that all collections existing
+      # before the addition of collection types do not need to be migrated.  They are assumed to be the default.
+      # @param collection [Collection] an instance of the Collection model
+      def load_collection_type_instance(collection)
+        if collection.collection_type_gid.nil?
+          collection.collection_type = Hyrax::CollectionType.find_or_create_default_collection_type
+          collection.collection_type_gid = collection.collection_type.gid
+          collection.save # do this one time on the first read by saving the gid in the collection object
+        else
+          collection.collection_type = Hyrax::CollectionType.find_by_gid(collection.collection_type_gid)
+          # TODO: What to do if collection_type isn't valid?
+          logger.warn("WARNING: Failed to load collection type instance.  Collection type is invalid.") unless collection_type_valid?(collection)
+        end
+      end
+
+      # Validate that the collection has a valid instance of the collection type.
+      # @param collection [Collection] an instance of the Collection model
+      # @return true if valid; otherwise, false
+      def collection_type_valid?(collection)
+        return false if collection.collection_type.nil?
```

Comment:
Should the collection's collection_type be checked for persistence? I believe that appears to be a requirement.